src/WebJobs.Script.WebHost/AssemblyAnalysis/AssemblyAnalysisService.cs

```diff
@@ -26,50 +26,64 @@ internal class AssemblyAnalysisService : IHostedService, IDisposable
         private Task _analysisTask;
         private bool _disposed;
         private bool _analysisScheduled;
+        private ILogger _logger;
 
         public AssemblyAnalysisService(IEnvironment environment, WebJobsScriptHostService scriptHost, ILoggerFactory loggerFactory, IOptionsMonitor<StandbyOptions> standbyOptionsMonitor)
         {
             _environment = environment;
             _scriptHost = scriptHost;
             _loggerFactory = loggerFactory;
             _standbyOptionsMonitor = standbyOptionsMonitor;
+            _logger = _loggerFactory.CreateLogger<AssemblyAnalysisService>();
         }
 
         public Task StartAsync(CancellationToken cancellationToken)
         {
-            if (!_environment.IsCoreTools())
+            try
             {
-                if (_standbyOptionsMonitor.CurrentValue.InStandbyMode)
+                if (!_environment.IsCoreTools())
                 {
-                    _standbyOptionsMonitor.OnChange(standbyOptions =>
+                    if (_standbyOptionsMonitor.CurrentValue.InStandbyMode)
                     {
-                        if (!standbyOptions.InStandbyMode && !_analysisScheduled)
+                        _standbyOptionsMonitor.OnChange(standbyOptions =>
                         {
-                            ScheduleAssemblyAnalysis();
-                        }
-                    });
-                }
-                else
-                {
-                    ScheduleAssemblyAnalysis();
+                            if (!standbyOptions.InStandbyMode && !_analysisScheduled)
+                            {
+                                ScheduleAssemblyAnalysis();
+                            }
+                        });
+                    }
+                    else
+                    {
+                        ScheduleAssemblyAnalysis();
+                    }
                 }
             }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error starting Assembly analysis service. Handling error and continuing.");
```

Comment:
nit: do we need to include `handling error and continuing` in the error message? I don't see where we're handling here, just looks like the ex is logged and swallowed?

Reply:
I was just trying to follow Brett's [changes ](https://github.com/Azure/azure-functions-host/commit/94f8767b7e20b11bb110e917bfd450c65b36f9a4#:~:text=_logger.LogError(ex%2C%20%22Error%20starting%20Rpc%20Initialization%20Service.%20Handling%20error%20and%20continuing.%22)%3B)from [this commit](https://github.com/Azure/azure-functions-host/commit/94f8767b7e20b11bb110e917bfd450c65b36f9a4) for consistency on handling IHostedServices based on the information given on [this issue](https://github.com/Azure/azure-functions-host/issues/4891). Let me know if I missed anything.